src/management-system-v2/lib/data/legacy/iam/users.ts

```diff
@@ -72,19 +75,43 @@ export function addUser(inputUser: OptionalKeys<User, 'id'>) {
   return user as User;
 }
 
-export function deleteuser(userId: string) {
+export class UserHasToDeleteOrganizationsError extends Error {
+  conflictingOrgs: string[];
+
+  constructor(conflictingOrgs: string[], message?: string) {
+    super(message ?? 'User has to delete organizations before being deleted');
+    this.name = 'UserHasToDeleteOrganizationsError';
+    this.conflictingOrgs = conflictingOrgs;
+  }
+}
+export function deleteUser(userId: string) {
   const user = usersMetaObject[userId];
 
   if (!user) throw new Error("User doesn't exist");
 
-  for (const environmentId of Object.keys(environmentsMetaObject)) {
-    if (environmentsMetaObject[environmentId].ownerId === userId) deleteEnvironment(environmentId);
+  const userOrganizations = getUserOrganizationEnviroments(userId);
```

Comment:
Not part of this PR, but just noticed the typo here too :D

Reply:
Yeah, I seem to be good at missing those :sweat_smile: